hbase-server/src/main/resources/hbase-webapps/master/hbck.jsp

```diff
@@ -37,9 +37,17 @@
 <%@ page import="org.apache.hadoop.hbase.util.Pair" %>
 <%@ page import="org.apache.hadoop.hbase.master.CatalogJanitor" %>
 <%@ page import="org.apache.hadoop.hbase.master.CatalogJanitor.Report" %>
+<%@ page import="org.apache.hadoop.hbase.util.Threads" %>
 <%
+  final String cacheParameterValue = request.getParameter("cache");
+  boolean cache = Boolean.valueOf(cacheParameterValue);
   final HMaster master = (HMaster) getServletContext().getAttribute(HMaster.MASTER);
   pageContext.setAttribute("pageTitle", "HBase Master HBCK Report: " + master.getServerName());
+  if (!cache) {
+    // Run the two reporters inline w/ drawing of the page. If exception, will show in page draw.
+    master.getMasterRpcServices().runHbckChore(null, null);
+    master.getMasterRpcServices().runCatalogScan(null, null);
```

Comment:
MasterRpcServices throw `ServiceException` for both `runHbckChore` and `runCatalogScan`. Maybe we can handle it and print some nice error message?

Reply:
Yeah. Was thinking about it. Though the rare SE not the end of the world. Added something at your prompting.... it looks like this when generated...

```
   if (!Boolean.parseBoolean(cacheParameterValue)) {
     // Run the two reporters inline w/ drawing of the page. If exception, will show in page draw.
     try {
       master.getMasterRpcServices().runHbckChore(null, null);
     } catch (org.apache.hbase.thirdparty.com.google.protobuf.ServiceException se) {
       out.write("Failed generating a new hbck_chore report; using cache; try again or run hbck_chore_run in the shell: " + se.getMessage() + "\n");
     }
     try {
       master.getMasterRpcServices().runCatalogScan(null, null);
     } catch (org.apache.hbase.thirdparty.com.google.protobuf.ServiceException se) {
       out.write("Failed generating a new catalogjanitor report; using cache; try again or run catalogjanitor_run in the shell: " + se.getMessage() + "\n");
     }
   }
```